### PR TITLE
Enable code generation splitting work on mgmt pipeline

### PR DESCRIPTION
**Pull request by sima-zhu**
The PR is to enable code check for all packages depend on resourcemananger. We will use resource manager pipeline to have the coverage.

Testing pipelines are associate in PR validation.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/scripts/CodeChecks.ps1 [hunk 1]
```diff
@@ -85,87 +88,90 @@ try {
...
+        Write-Host "Re-generating listings"
+        Invoke-Block {
+            & $PSScriptRoot\Export-API.ps1 -ServiceDirectory $ServiceDirectory -SDKType $SDKType -SpellCheckPublicApiSurface:$SpellCheckPublicApiSurface -ProjectListOverrideFileName $ProjectListOverrideFileName
+        }
     }
+    # The script is taking two path: $ServiceDirectory and $ProjectListOverrideFileName.
+    # If $ProjectListOverrideFileName exists, we don't validate the following steps in seperate code generation jobs.
+    # If $ProjectListOverrideFileName doesn't exist, we will check all steps under the $ServiceDirectory.
+    if (!$ProjectListOverrideFileName) {
+        Write-Host "Re-generating snippets"
```
Comment (sima-zhu) [inline eng/scripts/CodeChecks.ps1:108]: Nothing changed but alignment.

File: eng/scripts/CodeChecks.ps1 [hunk 2]
```diff
@@ -181,7 +187,7 @@ try {
     run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). `
```
Comment (sima-zhu) [inline eng/scripts/CodeChecks.ps1:187]: The git diff check applies for both $ServiceDirectory and $ProjectListOverrideFileName. No condition here.

File: eng/scripts/Update-Snippets.ps1 [hunk 3]
```diff
@@ -11,13 +11,7 @@ param (
...
 )
 
-$root = "$PSScriptRoot/../../sdk"
-
-# special casing * here because single invocation of SnippetGenerator is much faster than
-# running it per service directory
-if ($ServiceDirectory -and ($ServiceDirectory -ne "*")) {
-    $root += '/' + $ServiceDirectory
-}
+$root = "$PSScriptRoot/../../sdk/$ServiceDirectory"
```
Comment (sima-zhu) [inline eng/scripts/Update-Snippets.ps1:14]: Update snippets script doesn't run on everything anymore, so simplify the logic here.

File: eng/pipelines/templates/jobs/ci.yml [hunk 4]
```diff
@@ -215,6 +218,39 @@ jobs:
...
+        - template: /eng/pipelines/templates/steps/dependency.tests.yml
+          parameters:
+            TestDependsOnDependency: ${{ parameters.CodeCheckOnDependency }}
+            # The value for ProjectListOverrideFilePropertyName should be the same as AdditionalParameters ProjectListOverrideFilePropertyName below
+            ProjectListOverrideFilePropertyName: 'CodeGenerationProjectListFile'
+            ProjectFilesOutputFolder: $(Build.ArtifactStagingDirectory)
+            ExcludeTargetTestProjects: true
+            ServiceDirectory: ${{ parameters.ServiceDirectory }}
+            MatrixConfigsFile: eng/pipelines/templates/stages/code-gen-matrix.json
+            IncludeSrc: true
```
Comment (sima-zhu) [inline eng/pipelines/templates/jobs/ci.yml:244]: Include src and exclude tests for current scope. It is an ongoing conversation with mgmt team

File: eng/pipelines/templates/jobs/code.gen.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- name: ProjectListOverrideFilePropertyName
+  type: string
+  default: ''
+- name: ServiceDirectory
+  type: string
+  default: ''
+
+jobs:
+  - job:
+    displayName: "Code Generation"
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/code.gen.yml:30]: Based on our conversation we might not want to limit this to just Code Generation. We might want to make it a little more generic.

**Pull request closed without merging**